thunder/tests/distributed/test_ddp.py

```diff
@@ -1021,6 +1037,169 @@ def fwd_loss(m, x):
+                c10d.all_reduce(p)
+        cm.wait()
+
+        ref_state_dict = ref_model.state_dict()
+        expected = ref_model(x)
+
+        converter = {
+            "column": convert_module_to_columnwise_parallel,
+            "row": convert_module_to_rowwise_parallel,
+        }[name]
+        model = Model().to(device)
+        model.load_state_dict(ref_state_dict)
+        jitted_model = thunder.jit(model)
+        tp_jitted_model = converter(
+            jitted_model,
+            target_modules=("embed",),
+            process_group=process_group,
+        )
+        y = tp_jitted_model(x)
+
+        dim: int
+        orig_size: int
+        if name == "column":
+            dim = 0
+            orig_size = num_embeddings
+        else:
+            dim = 1
+            orig_size = embedding_dim
+        torch.testing.assert_close(
+            tp_jitted_model.get_parameter("embed.weight").size(dim), orig_size // self.world_size
+        )
+        torch.testing.assert_close(expected=expected, actual=y)
+
+        expected.mean().backward()
+        y.mean().backward()
+
+        torch.testing.assert_close(
+            expected=ref_model.embed.weight.grad.chunk(self.world_size, dim)[self.rank],
+            actual=tp_jitted_model.get_parameter("embed.weight").grad,
+        )
+
+    # TODO(crcrpar): Activate numerical check
+    @pytest.mark.skipif(torch.cuda.device_count() < 2, reason="")
+    def test_tensor_parallel_both_column_and_row(self):
+        num_embeddings = 128
+        embedding_dim = 32
+        n_hidden = 96
+        n_out = 16
+
+        class Model(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.embed_1 = nn.Embedding(num_embeddings, embedding_dim)
+                self.embed_2 = nn.Embedding(num_embeddings, embedding_dim)
+                self.linear1_0 = nn.Linear(embedding_dim, n_hidden)
+                self.linear1_1 = nn.Linear(n_hidden, n_hidden)
+                self.linear2_0 = nn.Linear(n_hidden, n_hidden)
+                self.linear2_1 = nn.Linear(n_hidden, n_out)
+
+            def forward(self, x):
+                feat_1 = self.embed_1(x)
+                feat_2 = self.embed_2(x)
+                sum_of_feat = feat_1 + feat_2
+                h = self.linear1_1(self.linear1_0(sum_of_feat))
+                return self.linear2_1(self.linear2_0(h))
+
+        device = torch.device("cuda", self.rank)
+        x = torch.randint(0, num_embeddings - 1, (16, 16), device=device)
+
+        process_group = None
+        ref_model = Model().to(device)
+        ref_state_dict = ref_model.state_dict()
+        expected = ref_model(x)
+
+        model = Model().to(device)
+        model.load_state_dict(ref_state_dict)
+        jitted_model = thunder.jit(model)
+
+        column_parallels = ["embed_1", "linear1_0", "linear2_1"]
+        row_parallels = ["embed_2", "linear1_1", "linear2_0"]
+        tp_jitted_model = convert_module_to_rowwise_parallel(
```

Comment:
just a comment: in the future it would be great to have high-level logic to dispatch row parallel and column parallel based on how ops on the trace are laid out.